### ISSUE TITLE
Support corecursive computation.

### ISSUE THREAD

**Issue by xekoukou** [agda/agda-ocaml#6]
Defn provides the necessary [info](https://github.com/agda/agda/blob/04941fb75be06039b0085ab047117625294cbc99/src/full/Agda/TypeChecking/Monad/Base.hs#L1601) to delay computation and force computation at the appropriate places with the [Lazy]( https://caml.inria.fr/pub/docs/manual-ocaml/libref/Lazy.html) module.

**Comment by fredefox** [comment]
It's great to have a more profficient ocamler on board!

**Comment by xekoukou** [comment]
Not really, I am 2 days into OCaml. It is just that Idris's [IR]( https://github.com/idris-lang/Idris-dev/blob/dbe521ff74189df85121abe454f86894de7fd75c/src/IRTS/Lang.hs#L32) has an ```LForce``` term and a ```LLazyApp``` term. 

Maybe @UlfNorell could help us by telling us where to place Force and Lazy ? There are 2 places with info. In the definition and in [TCase](https://github.com/agda/agda/blob/323f58f9b8dad239142ed1dfa0c60338ea2cb157/src/full/Agda/Syntax/Treeless.hs#L48)

**Comment by xekoukou** [comment]
I just looked at the treeless representation of the test [Issue2879-2](https://github.com/agda/agda/blob/323f58f9b8dad239142ed1dfa0c60338ea2cb157/test/Compiler/simple/Issue2879-2.agda)


```
=============== Issue2879-2.Delay′.force ===============
Treeless (abstract syntax)
  TLam (TCase 0 (CaseInfo {caseLazy = True, caseType = CTData Issue2879-2.Delay′}) (TError TUnreachable) [TACon {aCon = Issue2879-2.recCon-NOT-PRINTED, aArity = 1, aBody = TVar 0}])
```

```caseInfo``` provides the information that this is a lazy evalutation. One could use Ocaml's lazy operator here.

The question is where to force the lazy function to produce a term. Given the info on (mutual) recursion which we have already computed, I think that we need to put ```force``` when a function is called from within a non-mutually recursive function.
